src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -199,24 +200,89 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
                     (implicit actorRefFactory: ActorRefFactory, executionContext: ExecutionContext): Future[ObjectMetadata] = {
-    val metadataRequest = Get( getObjectResourceUrl(bucketName, objectKey) )
-    val metadataPipeline = addCredentials(OAuth2BearerToken(authToken)) ~> sendReceive ~> unmarshal[ObjectMetadata]
-    val request = metadataPipeline{metadataRequest}
 
-    request.recover {
+    // explicitly use the Google Cloud Storage xml api here. The json api requires that storage apis are enabled in the
+    // project in which a pet service account is created. The XML api does not have this limitation.
+    val metadataRequest = Head( getXmlApiMetadataUrl(bucketName, objectKey) )
+    val metadataPipeline = addCredentials(OAuth2BearerToken(authToken)) ~> sendReceive
+    metadataPipeline{metadataRequest} map { resp: HttpResponse =>
+      resp.status match {
+        case OK => xmlApiResponseToObject(resp, bucketName, objectKey)
+        case _ =>
+          // translate spray status code to akka-http status code
+          val code = akka.http.scaladsl.model.StatusCodes.getForKey(resp.status.intValue).getOrElse(
+            akka.http.scaladsl.model.StatusCodes.custom(resp.status.intValue, resp.status.defaultMessage)
+          )
+          throw new FireCloudExceptionWithErrorReport(ErrorReport(code, code.reason))
+      }
+    } recover {
       case t: UnsuccessfulResponseException =>
         throw new FireCloudExceptionWithErrorReport(FCErrorReport(t.response))
     }
   }
 
+  def getXmlApiMetadataUrl(bucketName: String, objectKey: String) = {
+    // explicitly use the Google Cloud Storage xml api here. The json api requires that storage apis are enabled in the
+    // project in which a pet service account is created. The XML api does not have this limitation.
+    val gcsStatUrl = "https://storage.googleapis.com/%s/%s"
+    gcsStatUrl.format(bucketName, java.net.URLEncoder.encode(objectKey,"UTF-8"))
+  }
+
+  def xmlApiResponseToObject(response: HttpResponse, bucketName: String, objectKey: String): ObjectMetadata = {
+    // crc32c and md5hash are both in x-goog-hash, which exists multiple times in the response headers.
+    val crc32c:String = response.headers
+      .find{h => h.lowercaseName.equals("x-goog-hash") && h.value.startsWith("crc32c=")}
+      .map(_.value.replaceFirst("crc32c=",""))
+        .getOrElse("")
+    val md5Hash: Option[String] = response.headers
+      .find{h => h.lowercaseName.equals("x-goog-hash") && h.value.startsWith("md5=")}
+      .map(_.value.replaceFirst("md5=",""))
+
+    val headerMap: Map[String, String] = response.headers.map { h =>
+      h.lowercaseName -> h.value
+    }.toMap
+
+    // exists in xml api, same value as json api
+    val generation: String = headerMap("x-goog-generation")
+    val size: String = headerMap("x-goog-stored-content-length") // present in x-goog-stored-content-length vs. content-length
```

Comment:
how did you choose from these 2?  Same q for `contentEncoding` since you chose the other one